### DISCUSSION TITLE
huskyが動いていない？

### DISCUSSION THREAD

**Discussion by kagomen** [lgtm-factory/lgtm-factory#157] [Scrap]
テストブランチを作成して、実際にcommitを試してみました

---

## コミット前の状態
- `npm run format`を実行するとフォーマットがかかってしまうnew_lgtm_design.mdファイルを複製した状態
- オリジナルが左、複製したものが右
<img width="1222" alt="スクリーンショット 2024-08-30 20 37 10" src="https://github.com/user-attachments/assets/ae2891b1-6e2f-470e-a0ee-27f817327128">

## コミット
- `git add .` `git commit -m "husky test"`を実行
- すると、複製したファイル（stagedファイル）に強制フォーマットが走り、オリジナルと行間隔が異なる
<img width="1218" alt="スクリーンショット 2024-08-30 20 38 00" src="https://github.com/user-attachments/assets/ac973618-9a19-4393-b303-4717865dff15">

## ためしに……
- `npm run format`を実行する
- オリジナルにフォーマットがかかり、複製したものにはフォーマットがかからない
<img width="1221" alt="スクリーンショット 2024-08-30 20 38 38" src="https://github.com/user-attachments/assets/45d2dd75-3c0e-4dd0-8b5e-59d5ccc56b63">

**Comment by kagomen** [comment]
## 推測

今回フォーマットがかからなかったファイルは、huskyがインストールされてなかった時（ #115 以前）のファイルだった・・・？
もしくは、#115 後に`npm install`を実行していない……？🤔

**Reply by kagomen** [reply]
## どうするか

とりあえず、現状のまま様子見したいです（設定を間違えている感じがしないので！）
また同じようなことが起これば、Discordで話した、

> **/*指定をやめて、npm run formatと同じコマンド（`prettier --write`）を指定する

で対応できればと思います……！！！

**Reply by kazzyfrog** [reply]
> とりあえず、現状のまま様子見

了解です👍

**Comment by kazzyfrog** [comment]
あれ、単純に、コマンドが統一されていないことが原因で、
huskyと`npm run format`の整形が違うのかな？と思いましたが、そうでもない感じですかね！

「--ignore-unknown オプション」があるかないかは、別に同じですかね。。？

**Reply by kagomen** [reply]
ですね……。
--ignore-unknownオプションは、ただPrettierが対応していないファイルをスルーするだけのコマンドなので、

Discordに書いたんですが、

> **/*で全ディレクトリを指定してるので、 --ignore--unknownオプションつけないとエラーで中止しちゃう

んですよね
以下、lint-stagedコマンドを`"**/*": "prettier --write"`に変更して、svgファイルを追加し、git addとcommitした時のコマンドです

<img width="613" alt="スクリーンショット 2024-08-30 23 21 59" src="https://github.com/user-attachments/assets/55961911-ff0c-482e-9a18-8633ac09589c">


簡潔にまとめると、`npm run format`に登録している`"prettier --write ."`と` "**/*": "prettier --write --ignore--unknown"`は、同義なはず……！

なので、もうしばらく様子見させてくださいー！
